### PR TITLE
Allows Passing Custom Chip Children

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Screen Shot 2020-03-17 at 3 10 47 PM

### What does this PR do, or why is it needed?

See #1392 

### What design trade-offs/decisions were made?

Deprecated `icon` prop

### How do I test this PR?

Stories

### What automated tests did you write?

Snaps

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-kit/src/Chip/Chip.stories.js [hunk 1]
```diff
@@ -37,4 +39,17 @@ storiesOf('ui-kit/Chip', module)
...
+  ))
+  .add('Custom children', () => (
+    <Chip onPress={() => {}}>
+      <Text>Custom children</Text>
+    </Chip>
+  ))
+  .add('Custom icon', () => (
+    <Chip onPress={() => {}} title="Custom Icon">
+      {/* eslint-disable-next-line */}
+      <View style={{ paddingRight: 8 }}>
```
Comment (conrad-vanl) [inline packages/apollos-ui-kit/src/Chip/Chip.stories.js]: @burkeshartsis only problem I see with using this `children` props approach is that you now have to know how to pad your icon to show up right

Comment (burkeshartsis) [reply packages/apollos-ui-kit/src/Chip/Chip.stories.js]: I agree but if you are a developer using our `ui-kit` at this level this feels like a reasonable ask. The developer is already importing `Chip` and positioning it on the screen. I don't think it's crazy to assume that if they pass an `Icon` via `children` that they would see the need for some padding and know how to accomplish that. Plus we don't even know how much padding they need. This gives the user more control and at this level of the platforms api that feels right.

Comment (redreceipt) [reply packages/apollos-ui-kit/src/Chip/Chip.stories.js]: We could just give the title a bit of `paddingHorizontal`? similar to before but it would keep it centered

Comment (redreceipt) [reply packages/apollos-ui-kit/src/Chip/Chip.stories.js]: or even better, we could give the whole view `paddingHorizontal` and then `flex: space-between` the elements

Comment (burkeshartsis) [reply packages/apollos-ui-kit/src/Chip/Chip.stories.js]: adding `paddingHorizontal` without a condition would make the chip "wider" if there were no other children. This would change its default "shape." 

You could conditionally add padding to `TitleText` if there were `children` but, as I mentioned above, you would be making a blind assumption as to how much padding you needed.

Comment (burkeshartsis) [reply packages/apollos-ui-kit/src/Chip/Chip.stories.js]: The other option here, which @conrad-vanl suggested early on is to use a `Icon` prop which takes a component. By being more specific you can make stronger assumptions like padding the text. The down side to this, outside what I have mentioned already, is that we already have an established api pattern for this with `Button`. I'd just assume keep it consistent.

**Comment by redreceipt** [comment]
Up to you @burkeshartsis to merge as is or tell me to change something.
I'll do whatever.


--
Michael Neeley
Junior Web Developer
NewSpring Church
www.newspring.cc
Cell: (803) 622-6599

**Comment by redreceipt** [comment]
@burkeshartsis what's the decision here?
